chess2.cpp

```diff
@@ -0,0 +1,882 @@
+    str mystr = to_string(row) + to_string(col) + to_string(desrow) + to_string(descol);
+    return mystr;
+}
+
+str Chess2::computer_move(unsigned short int depth) {
+    str bestmove;
+    bestmove = miniMaxroot(depth, false);
+    perform(bestmove);
+    return bestmove;
+}
+
+buff Chess2::getallmoves(bool turn) {
+    int original = 0;
+    pseudomoves.clear();
+    legal_moves.clear();
+    if (turn == true) {
+        for (int i = 0; i < 8; i++) {
+            for (int j = 0; j < 8; j++) {
+                if (!board[i][j]) continue;
+                if (board[i][j] == 1) getpawnmoves(true, i, j);
+                else if (board[i][j] == 2) getdiagonalmoves(true, i, j);
+                else if (board[i][j] == 3) getknightmoves(true, i, j);
+                else if (board[i][j] == 5) getstraigtmoves(true, i, j);
+                else if (board[i][j] == 6) {
+                    getdiagonalmoves(true, i, j);
+                    getstraigtmoves(true, i, j);
+                }
+                else if (board[i][j] == 10) getkingmoves(true, i, j);
+            }
+        }
+        for(std::string i:pseudomoves){
+            original = perform(i);
+            if (check(true) == false) {
+                legal_moves.push_back(i);
+            }
+            undomove(original, i);
+        }
+    }
+    else if (!turn) {
+        for (int i = 0; i < 8; i++) {
+            for (int j = 0; j < 8; j++) {
+                if (!board[i][j]) continue;
+                else if (board[i][j] == -1){
+                    getpawnmoves(false, i, j);
+                }
+                else if (board[i][j] == -2) getdiagonalmoves(false, i, j);
+                else if (board[i][j] == -3) getknightmoves(false, i, j);
+                else if (board[i][j] == -5) getstraigtmoves(false, i, j);
+                else if (board[i][j] == -6) {
+                    getdiagonalmoves(false, i, j);
+                    getstraigtmoves(false, i, j);
+                }
+                else if (board[i][j] == -10) getkingmoves(false, i, j);
+            }
+        }
+        for(std::string i:pseudomoves){
+            original = perform(i);
+            if (check(false) == false) {
+                legal_moves.push_back(i);
+            }
+            undomove(original, i);
+        }
+    }
+    return legal_moves;
+}
+
+bool Chess2::check(bool turn) {
+
+    if (turn == true) {
+        bool found = false;
+        int row, col;
+        //Finding the king on the board
+
+        for (int i = 0; i < 8; i++) {
+            for (int j = 0; j < 8; j++) {
+                if (board[i][j] == 10) {
+                    row = i;
+                    col = j;
+                    found = true;
+                    break;
+                }
+            }
+        }
+        if (found == false){
+            return false;
+        }
+
+        //Finding the king on the board
+        if (row != 0 && col != 0 && board[row - 1][col - 1] == -1) return true;
```

Comment:
Divide this check function into smaller functions like check_pawn(row, col, turn), check_rook_queen(row, col, turn)